docs/PremiumApps.md

```diff
@@ -0,0 +1,488 @@
+          scope: premiumApps             <-------- setting scope as premiumApps to install ES
+          premiumAppsProps:
+            type: enterpriseSecurity     <-------- installing premiumApp Splunk Enterprise Security. 
+            esDefaults:
+              sslEnablement: ignore       <--------- setting ssl_enablement to ignore
+```
+
+##### Search Head Cluster server.conf timeouts
+
+It may be necessary to increase the value of the default Search Head Clustering network timeouts to ensure that the connections made from the deployer to the Search Heads while pushing apps do not timeout. 
+
+These timeouts can be set through defaults.yaml
+```yaml
+  defaults: |-
+    splunk:
+      conf:
+        - key: server
+          value:
+            directory: /opt/splunk/etc/system/local
+            content:
+              shclustering:         
+                rcv_timeout: 300
+                send_timeeout: 300
+                cxn_timeeout: 300
+```
+
+##### splunkdConnectionTimeout
+Increasing the value of splunkdConnectionTimeout in the web.conf will help ensure that all API calls made by the installer script will not timeout and prevent the installation from succeeding.
+```yaml
+  defaults: |-
+    splunk:
+      conf:
+        - key: web
+          value:
+            directory: /opt/splunk/etc/system/local
+            content:
+              settings:
+                splunkdConnectionTimeout: 300
+```
+
+### Example YAMLs
+
+In the following examples, you can find the YAML files for installing ES in a Splunk Standalone, Splunk SHC, and Indexer Cluster.
+
+#### Install ES on a standalone splunk deployment
+
+#### Install ES on a standalone splunk deployment with sslEnablement=ignore
+
+Here is an example of a standalone CR with the `sslEnablement=ignore`
+
+Assumption: the ES app tarball exists in an s3 bucket folder named "es_app" under the parent folder "security-team-apps" in your s3 bucket.
+
+```yaml
+apiVersion: enterprise.splunk.com/v4
+kind: Standalone
+metadata:
+  name: example
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  replicas: 1
+  appRepo:
+    appsRepoPollIntervalSeconds: 60
+    defaults:
+      volumeName: volume_app_repo
+      scope: local
+    appSources:
+      - name: esApp
+        location: es_app/
+        scope: premiumApps
+        premiumAppsProps:
+          type: enterpriseSecurity
+          esDefaults:
+             sslEnablement: ignore
+    volumes:
+      - name: volume_app_repo
+        storageType: s3
+        provider: aws
+        path: security-team-apps/
+        endpoint: https://s3-us-west-2.amazonaws.com
+        region: us-west-2
+        secretRef: splunk-s3-secret
+```
+
+#### Install ES on a standalone splunk deployment with sslEnablement value strict
+
+Here is an example of a standalone CR with the `sslEnablement=strict`. 
+
+Assumption: the ES app tarball exists in an s3 bucket folder named "es_app" under the parent folder "security-team-apps" in your s3 bucket.
```

Comment:
Should we add a comment here saying that we use the env variable `SPLUNK_HTTP_ENABLESSL` to enable SSL on splunkd?

Reply:
thanks. added in some form.